cpp/src/parquet/column_writer.cc

```diff
@@ -1332,13 +1337,38 @@ class TypedColumnWriterImpl : public ColumnWriterImpl, public TypedColumnWriter<
       bits_buffer_->ZeroPadding();
     }
 
-    if (leaf_array.type()->id() == ::arrow::Type::DICTIONARY) {
-      return WriteArrowDictionary(def_levels, rep_levels, num_levels, leaf_array, ctx,
-                                  maybe_parent_nulls);
+    if (this->properties_->cdc_enabled()) {
+      ARROW_ASSIGN_OR_RAISE(auto boundaries,
+                            content_defined_chunker_.GetBoundaries(
+                                def_levels, rep_levels, num_levels, leaf_array));
+      for (auto chunk : boundaries) {
+        auto chunk_array = leaf_array.Slice(chunk.value_offset);
```

Comment:
It seems that we need to specify not only offset but also length. If we don't specify length, we may use duplicated data when `leaf_array` is split to 3 or more chunks.

Reply:
Theoretically the same value index advancing logic is used in the writer implementations. 
I can track the length of each value chunk just in case though. Can you think of an example which I could test against?